pkg/pillar/cmd/zedkube/applogs.go

```diff
@@ -264,3 +269,31 @@ func getnodeNameAndUUID(ctx *zedkubeContext) error {
 	}
 	return nil
 }
+
+func handleKubePodsGetError(ctx *zedkubeContext, items, stItems map[string]interface{}) {
+	if ctx.getKubePodsError.getKubePodsErrorTime == nil {
+		now := time.Now()
+		ctx.getKubePodsError.getKubePodsErrorTime = &now
+		log.Noticef("handleKubePodsGetError: can't get pods for %s, ", aiconfig.DisplayName)
+	} else if time.Since(*ctx.getKubePodsError.getKubePodsErrorTime) > 2*time.Minute {
+		if ctx.getKubePodsError.processedErrorCondition == false {
+			ctx.getKubePodsError.processedErrorCondition = true
+			for _, item := range items {
+				aiconfig := item.(types.AppInstanceConfig)
+				for _, st := range stItems {
+					aiStatus := st.(types.ENClusterAppStatus)
+					if aiStatus.AppUUID == aiconfig.UUIDandVersion.UUID {
+						// if we used to publish the status, of this app is scheduled on this node
+						// need to reset this, since we have lost the connection to the kubernetes
+						// for longer time than the app is to be migrated to other node
+						if aiStatus.ScheduledOnThisNode {
+							aiStatus.ScheduledOnThisNode = false
+							ctx.pubENClusterAppStatus.Publish(aiconfig.Key(), aiStatus)
```

Comment:
Do we need to set Dnid too ? Or does it not matter ? Just for correctness

Reply:
Should apply in all the cases, any node, regardless of DNID, if previously is scheduled for running, and now it's not k3s reachable, we set this to remove the flag.